core/rx-java-driver/src/main/java/org/hawkular/rx/cassandra/driver/RxSessionImpl.java

```diff
@@ -38,9 +43,14 @@
 public class RxSessionImpl implements RxSession {
 
     private Session session;
+    private int maximumInflightRequests;
 
     public RxSessionImpl(Session session) {
         this.session = session;
+        maximumInflightRequests = Math.min(
```

Comment:
I am not sure I understand. ```maximumInFlightRequests``` is set to the smaller of the max local or remote connections, but each connection supports concurrent requests. Why is the max requests per connection not taken into account?

>We don't know what host is being targeted

Actually we can easily determine that with ```LoadBalancingPolicy.newQueryPlan(String loggedKeyspace, Statement statement)``` but I do not know that it is necessary. I think it is sufficient to see if there is an available connection with **any** host. I say this because we use a TokenAwarePolicy with a child load balancing policy of DCAwareRoundRobinPolicy. Even if resources are not available to send the query directly to the replica, I believe the driver will fall back to a non-replica node.